app/invocation/invocation_action_card.tsx

```diff
@@ -0,0 +1,153 @@
+        this.setState({
+          ...this.state,
+          error: "Error loading command profile. Make sure your cache is correctly configured.",
+        });
+      });
+  }
+
+  displayList(list: string[]) {
+    if (list.length == 0) return <div>None found.</div>;
+    return <div className="action-list">{list.map((argument) => <div>{argument}</div> || [])}</div>;
+  }
+
+  render() {
+    return (
+      <div>
+        <div className="card">
+          <img className="icon" src="/image/info.svg" />
+          <div className="content">
+            <div className="title">Action details </div>
+            <div className="details">
+              {this.state.action && (
+                <div>
+                  <div className="action-section">
+                    <div className="action-property-title">Hash/Size: </div>
+                    <div>{this.props.search.get("actionDigest")}</div>
+                  </div>
+                  <div className="action-section">
+                    <div className="action-property-title">Output Node Properties: </div>
+                    {this.state.action.outputNodeProperties.length ? (
+                      <div>
+                        {this.state.action.outputNodeProperties.map((outputNodeProperty) => (
+                          <div className="output-node">{outputNodeProperty}</div>
+                        ))}
+                      </div>
+                    ) : (
+                      <div>None found.</div>
+                    )}
+                  </div>
+                  <div className="action-section">
+                    <div className="action-property-title">Do Not Cache: </div>
+                    <div>{this.state.action.doNotCache ? "True" : "False"}</div>
+                  </div>
+                </div>
+              )}
+            </div>
+          </div>
+        </div>
+        <div className="card">
+          <img className="icon" src="/image/info.svg" />
+          <div className="content">
+            <div className="title">Command details</div>
+            <div className="details">
+              {this.state.command && (
+                <div>
+                  <div className="action-section">
+                    <div className="action-property-title">Arguments:</div>
+                    {this.displayList(this.state.command.arguments)}
+                  </div>
+                  <div className="action-section">
+                    <div className="action-property-title">Environment Variables:</div>
+                    <div className="action-list">
+                      {this.state.command.environmentVariables.map(
+                        (variable) =>
+                          (
+                            <div>
+                              <span className="env-name">{variable.name}</span>
+                              <span className="env-value">={variable.value}</span>
+                            </div>
+                          ) || []
```

Comment:
I'm not sure this `|| []` bit does anything here, I think you can just remove it. Same above in `displayList`